src/clients/java/src/client.zig

```diff
@@ -1,62 +1,361 @@
+            _ = class;
+            var context = NativeClient.client_init(
+                false,
+                env,
+                @bitCast(u32, cluster_id),
+                addresses,
+                @bitCast(u32, max_concurrency),
+            );
+            return @bitCast(jni.JLong, @ptrToInt(context));
+        }
+
+        fn client_init_echo(
+            env: *jni.JNIEnv,
+            class: jni.JClass,
+            cluster_id: jni.JInt,
+            addresses: jni.JString,
+            max_concurrency: jni.JInt,
+        ) callconv(jni.JNICALL) jni.JLong {
+            _ = class;
+            var context = NativeClient.client_init(
+                true,
+                env,
+                @bitCast(u32, cluster_id),
+                addresses,
+                @bitCast(u32, max_concurrency),
+            );
+            return @bitCast(jni.JLong, @ptrToInt(context));
+        }
+
+        fn client_deinit(
+            env: *jni.JNIEnv,
+            class: jni.JClass,
+            context_handle: jni.JLong,
+        ) callconv(jni.JNICALL) void {
+            _ = env;
+            _ = class;
+            NativeClient.client_deinit(@intToPtr(*Context, @bitCast(usize, context_handle)));
+        }
+
+        fn submit(
+            env: *jni.JNIEnv,
+            class: jni.JClass,
+            context_handle: jni.JLong,
+            request_obj: jni.JObject,
+        ) callconv(jni.JNICALL) jni.JInt {
+            _ = class;
+            assert(context_handle != 0);
+            const packet_acquire_status = NativeClient.submit(
+                env,
+                @intToPtr(*Context, @bitCast(usize, context_handle)),
+                request_obj,
+            );
+
+            return @intCast(jni.JInt, @enumToInt(packet_acquire_status));
+        }
+    };
+
+    @export(Exports.on_load, .{ .name = "JNI_OnLoad", .linkage = .Strong });
+    @export(Exports.on_unload, .{ .name = "JNI_OnUnload", .linkage = .Strong });
+
+    @export(Exports.client_init, .{ .name = prefix ++ "clientInit", .linkage = .Strong });
+    @export(Exports.client_init_echo, .{ .name = prefix ++ "clientInitEcho", .linkage = .Strong });
+    @export(Exports.client_deinit, .{ .name = prefix ++ "clientDeinit", .linkage = .Strong });
+    @export(Exports.submit, .{ .name = prefix ++ "submit", .linkage = .Strong });
+}
+
+/// Reflection helper and metadata cache.
+const ReflectionHelper = struct {
+    var initialization_exception_class: jni.JClass = null;
+    var initialization_exception_ctor_id: jni.JMethodID = null;
+    var assertion_error_class: jni.JClass = null;
+
+    var request_class: jni.JClass = null;
+    var request_send_buffer_field_id: jni.JFieldID = null;
+    var request_send_buffer_len_field_id: jni.JFieldID = null;
+    var request_reply_buffer_field_id: jni.JFieldID = null;
+    var request_operation_method_id: jni.JMethodID = null;
+    var request_end_request_method_id: jni.JMethodID = null;
```

Comment:
I guess potentially we can do something like

```
const ReflectionHelper = struct {
    var instance: ?@This() = null;

    initialization_exception_class: jni.JClass,
    initialization_exception_ctor_id: jni.JMethodID,
    assertion_error_class: jni.JClass,

    request_class: jni.JClass,
    request_send_buffer_field_id: jni.JFieldID,
    request_send_buffer_len_field_id: jni.JFieldID,
    request_reply_buffer_field_id: jni.JFieldID,
    request_operation_method_id: jni.JMethodID,
    request_end_request_method_id: jni.JMethodID,
```

to get some compile-time checking that we try to initialize everything. But as jni.JClass and such are themselves nullable, I don't think it is worth it (that is, we need to check _both_ that we tried to initialized a field, and that `find_class` returned a non-null value).

Reply:
Since the type definition is `?*anyopaque`, I think it's just [feng shui programming](https://github.com/SerenityOS/serenity/blob/60ac254df62408bf1831d28c9a712ee3d27a1366/CONTRIBUTING.md?plain=1#L70) 🤣